src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -734,6 +740,95 @@ public void Purge(Document document)
             });
         }
 
+        /// <summary>
+        /// Purges the given document id of the <see cref="Document"/> 
+        /// from the database.  This leaves no trace behind and will 
+        /// not be replicated
+        /// </summary>
+        /// <param name="docId">The id of the document to purge</param>
+        /// <returns>Whether or not the document was actually purged.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when trying to purge a document 
+        /// from a database other than the one it was previously added to</exception>
+        [ContractAnnotation("null => halt")]
+        public void Purge(string docId)
+        {
+            CBDebug.MustNotBeNull(Log.To.Database, Tag, nameof(docId), docId);
+            var document = GetDocument(docId);
+            ThreadSafety.DoLocked(() =>
+            {
+                CheckOpen();
+                VerifyDB(document);
+
+                if (!document.Exists) {
+                    throw new CouchbaseLiteException(C4ErrorCode.NotFound);
+                }
+
+                InBatch(() =>
+                {
+                    var result = Native.c4doc_purgeRevision(document.c4Doc.RawDoc, null, null);
+                    if (result >= 0) {
+                        LiteCoreBridge.Check(err => Native.c4doc_save(document.c4Doc.RawDoc, 0, err));
+                    }
+                });
+
+                document.ReplaceC4Doc(null);
+            });
+        }
```

Comment:
Is the above a copy paste of another method?  Could the other method be called instead?

Reply:
This is based on https://github.com/couchbaselabs/couchbase-lite-apiv2/pull/218 #3.

Reply:
What I mean is that the body looks copy and pasted.  Could one method call the other?